store/src/main/java/org/apache/rocketmq/store/ha/HAService.java

```diff
@@ -287,17 +287,12 @@ private void doWaitTransfer() {
             if (!this.requestsRead.isEmpty()) {
                 for (CommitLog.GroupCommitRequest req : this.requestsRead) {
                     boolean transferOK = HAService.this.push2SlaveMaxOffset.get() >= req.getNextOffset();
-                    long waitUntilWhen = HAService.this.defaultMessageStore.getSystemClock().now()
-                            + HAService.this.defaultMessageStore.getMessageStoreConfig().getSyncFlushTimeout();
-                    while (!transferOK && HAService.this.defaultMessageStore.getSystemClock().now() < waitUntilWhen) {
+                    long deadLine = req.getDeadLine();
+                    while (!transferOK && deadLine - System.nanoTime() > 0) {
```

Comment:
replace 
```java
HAService.this.defaultMessageStore.getSystemClock().now() + req.getTimeoutMillis()
```
with deadLine is not necessary and recommended.

Reply:
nanoTIme() is better than currentTimeMillis() for measure elapsed time. currentTimeMillis() may affect by system clock  adjustment, and cause unexpected result.